src/utils.rs

```diff
@@ -707,25 +707,42 @@ impl<'a, 'b> fmt::Display for Emoji<'a, 'b> {
 }
 
 fn str_width(s: &str) -> usize {
-    #[cfg(feature = "unicode-width")]
+    #[cfg(feature = "unicode-display-width")]
     {
-        use unicode_width::UnicodeWidthStr;
-        s.width()
+        unicode_display_width::width(s) as usize
     }
-    #[cfg(not(feature = "unicode-width"))]
+    #[cfg(not(feature = "unicode-display-width"))]
     {
         s.chars().count()
     }
 }
 
 #[cfg(feature = "ansi-parsing")]
 fn char_width(c: char) -> usize {
-    #[cfg(feature = "unicode-width")]
-    {
-        use unicode_width::UnicodeWidthChar;
-        c.width().unwrap_or(0)
+    #[cfg(feature = "unicode-display-width")]
+    if c < '\u{7F}' {
```

Comment:
Why did you go this route? I'd like to avoid referring to explicit code points as much as possible as it would be likely for these to fall out of sync for Unicode updates -- that's what we're trying to rely on other crates for.